### PR TITLE
MBA fixes - AMD 

### PR DESCRIPTION
**Pull request by babumoger**
Couple of fixes.
1. MBA feature was broken for AMD
2. Changed printing messages from SOCKET to Core Complex for AMD
 
## Description

The MBA feature was not working when trying to set the MBA limits on AMD.
....
# pqos -I OS -e mba:1=0xf
NOTE:  Mixed use of MSR and kernel interfaces to manage
       CAT or CMT & MBM may lead to unexpected behavior.
ERROR: Expected MBA controller but not requested!
SOCKET 0 MBA COS1 - FAILED!
Allocation configuration error!
```
The code was using the hard coded value to detect the feature. Changed it to use the vendor specific mba_max values while detecting the feature.

Also fixed printing  messages on AMD.
Before the change.
SOCKET 0 MBA COS1 => 256% requested, 256% applied
SOCKET 1 MBA COS1 => 256% requested, 256% applied
SOCKET 2 MBA COS1 => 256% requested, 256% applied
SOCKET 3 MBA COS1 => 256% requested, 256% applied

After the change.

Core Complex 0 MBA COS1 => 256 requested, 256 applied
Core Complex 1 MBA COS1 => 256 requested, 256 applied
Core Complex 2 MBA COS1 => 256 requested, 256 applied
Core Complex 3 MBA COS1 => 256 requested, 256 applied

## Affected parts
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] library
- [x] pqos utility
- [ ] rdtset utility
- [ ] other: (please specify)

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

Without this fix MBA feature wont work on AMD.

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, tests ran to see how -->
<!--- your change affects other areas of the code, etc. -->
Tested on AMD system.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: lib/os_cap.c [hunk 1]
```diff
@@ -675,7 +680,7 @@ os_cap_get_mba_ctrl(const struct pqos_cap *cap,
                 if (ret == PQOS_RETVAL_OK) {
                         fd = resctrl_alloc_fopen(grp, "schemata", "w");
                         if (fd != NULL) {
-                                fprintf(fd, "MB:0=2000\n");
+                                fprintf(fd, "MB:0=%d\n", vconfig->mba_max * 2);
```
Comment (aleksinx) [inline lib/os_cap.c:683]: Compilation warning
`os_cap.c:683: warning: %d in format string (no. 1) requires 'int' but the argument type is 'unsigned int'.`
```suggestion
                                fprintf(fd, "MB:0=%u\n", vconfig->mba_max * 2);
```

File: pqos/alloc.c [hunk 2]
```diff
@@ -379,21 +387,32 @@ set_mba_cos(const unsigned class_id,
          * Set all selected classes
          */
         for (i = 0; i < sock_num; i++) {
+                const char *unit;
+                const char *package;
                 int ret = pqos_mba_set(sock_ids[i], 1, &mba, &actual);
 
+                if (cpu->vendor == PQOS_VENDOR_AMD) {
+                        package = "Core Complex ";
```
Comment (aleksinx) [inline pqos/alloc.c:395]: Please remove redundant space
```suggestion
                        package = "Core Complex";
```

File: pqos/alloc.c [hunk 3]
```diff
@@ -379,21 +387,32 @@ set_mba_cos(const unsigned class_id,
...
+                        unit = "";
+                } else {
+                        package = "SOCKET ";
```
Comment (aleksinx) [inline pqos/alloc.c:398]: Please remove redundant space
```suggestion
                        package = "SOCKET";
```

**Review by aleksinx — CHANGES_REQUESTED** [review]

File: pqos/alloc.c [hunk 4]
```diff
@@ -228,18 +230,23 @@ set_l3_cos(const unsigned class_id,
 
                 /* set new L3 class definition */
                 ret = pqos_l3ca_set(sock_ids[i], 1, &ca);
+                if (cpu->vendor == PQOS_VENDOR_AMD)
+                        package = "Core Complex ";
```
Comment (aleksinx) [inline pqos/alloc.c:234]:
```suggestion
                        package = "Core Complex";
```

File: pqos/alloc.c [hunk 5]
```diff
@@ -228,18 +230,23 @@ set_l3_cos(const unsigned class_id,
...
+                else
+                        package = "SOCKET ";
```
Comment (aleksinx) [inline pqos/alloc.c:236]:
```suggestion
                        package = "SOCKET";
```

**Review by aleksinx — CHANGES_REQUESTED** [review]

**Comment by babumoger** [comment]
Addressed all the comments.  Let me know if I missed something. Thanks

**Comment by babumoger** [comment]
Sent new pull request https://github.com/intel/intel-cmt-cat/pull/196 to address all the comments. I will close this pull request.

**Pull request closed without merging**